src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,278 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgGroup, ArgMatches, Command};
+use pid::{walk_pid, PidEntry, TerminalType};
+use std::{borrow::BorrowMut, cmp::Ordering, collections::HashSet};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+    let patterns = collect_arg_patterns(&matches);
+
+    let Some(result) =
+        collect_oldest_newest(&matches, &patterns).or(collect_normal_pid(&matches, &patterns))
+    else {
+        println!("pgrep: no matching criteria specified");
+        println!("Try `pgrep --help' for more information");
+        return Ok(());
+    };
+
+    // Just processsing output format here
+    let result = || {
+        if matches.get_flag("count") {
+            return format!("{}", result.len());
+        };
+
+        let delimiter = matches.get_one::<String>("delimiter").unwrap();
+
+        let formatted: Vec<_> = if matches.get_flag("list-full") {
+            result
+                .into_iter()
+                .map(|it| format!("{} {}", it.pid, it.cmdline))
+                .collect()
+        } else if matches.get_flag("list-name") {
+            result
+                .into_iter()
+                .map(|it| format!("{} {}", it.pid, it.clone().status().get("Name").unwrap()))
+                .collect()
+        } else {
+            result.into_iter().map(|it| format!("{}", it.pid)).collect()
+        };
+
+        formatted.join(delimiter)
+    };
+
+    let result = result();
+    if !result.is_empty() {
+        println!("{}", result);
+    };
+
+    Ok(())
+}
+
+fn collect_arg_patterns(matches: &ArgMatches) -> Vec<String> {
+    let should_ignore_case = matches.get_flag("ignore-case");
+
+    let programs = matches
+        .get_many::<String>("pattern")
+        .unwrap_or_default()
+        .map(|it| it.to_string());
+
+    if should_ignore_case {
+        programs.map(|it| it.to_lowercase()).collect::<Vec<_>>()
+    } else {
+        programs.collect()
+    }
+}
+
+fn collect_pid(matches: &ArgMatches, patterns: &[String]) -> Vec<PidEntry> {
```

Comment:
This function returns a collection and so my suggestion is to use the plural of "pid" in the function name: `collect_pids`.